modules/reports/models/scheduled_reports.php

```diff
@@ -234,18 +234,47 @@ static public function edit_report($id=false, $rep_type=false, $saved_report_id=
 	 */
 	static function update_report_field($id=false, $field=false, $value=false)
 	{
-		$db = Database::instance();
-		$id = (int)$id;
-		$field = $db->escape_column(trim($field));
-		$value = $db->escape(trim($value));
-		$sql = "UPDATE scheduled_reports SET {$field}={$value} WHERE id={$id}";
+		$allowed_columns = [
+			"recipients",
+			"description",
+			"filename",
+			"local_persistent_filepath",
+			"attach_description",
+		];
+		$column = array_search($field, $allowed_columns, TRUE);
+
+		if (!$column) {
+			return false;
+		}
+
+		$dbconn = Kohana::config('database.default');;
+		// Create connection
+		$conn = new mysqli(
+			$dbconn['connection']['host'],
+			$dbconn['connection']['user'],
+			$dbconn['connection']['pass'],
+			$dbconn['connection']['database']
+		);
+		// Check connection
+		if ($conn->connect_error) {
+			throw new Exception("Error connecting to datase");
+		}
+		// Prepared statement
+		$stmt = $conn->prepare("UPDATE scheduled_reports SET {$allowed_columns[$column]}=? WHERE id=?");
```

Comment:
Isn't $allowed_columns[$column] == $field ?

Reply:
It should be the same as field, yes. but since we dont want to include user input in our column select. We use a predetermined "safe" string. It should be safe to use $field here. But since we use a string comparison I would rather use a predetermined value. There could be security flaws in the future.